### PR TITLE
Supporting files for buildings theme

### PR DESCRIPTION
**Pull request by LukaszMichalski-TomTom**
## Feb 22, 2024: This PR has been resurrected and re-based, scroll to bottom for context

---

Added mapping files for OSM values to Overture enumeration values.
Additional values:
**facadeMaterial**:
- clay
- timber_framing

**roofMaterial**
- gravel
- solar_panels

**roofShape**:
- saltbox
- skillion
- sawtooth

Four roof shape names changed to OSM equivalents, e.g. `gable`→`gabled`. I did it to make it more in line with OSM tooling and possible renders that can generate roof geometry based on this value. If someone would like to use Overture data to generate visualization, it may be easier for them.

### CODE REVIEW

**Review by jwass — COMMENTED** [review]
@LukaszMichalski-TomTom I realize these aren't all your doing, but can you make all the Overture values camelCased?

It's the standard (even though I don't love it).

**Comment by jwass** [comment]
@LukaszMichalski-TomTom Just so I understand... facadeMaterial describes the mapping for 
* `building:material=*`  (and also `building:facade:material`?)
* roofMaterial is for `roof:material`
* roofShape is for `roof:shape`
??

**Review by skmoore — CHANGES_REQUESTED** [review]

**Comment by LukaszMichalski-TomTom** [comment]
> @LukaszMichalski-TomTom Just so I understand... facadeMaterial describes the mapping for
> 
> * `building:material=*`  (and also `building:facade:material`?)
> * roofMaterial is for `roof:material`
> * roofShape is for `roof:shape`
>   ??

I will add README next to mapping files to describe osm tag that need to be used. 
`overture_facadeMaterial_OSM_mapping.json` file contains mapping of OSM values to Overture `facedMaterial` property. OSM tags used for mapping in order:
- `building:material` (documentation: https://wiki.openstreetmap.org/wiki/Key:building:material)
- `building:facade:material` (documentation: https://wiki.openstreetmap.org/wiki/Key:building:facade:material)

`overture_roofMaterial_OSM_mapping.json` file contains mapping of OSM values to Overture `roofMaterial` property. OSM tags used for mapping:
- `roof:material` (documentation: https://wiki.openstreetmap.org/wiki/Key:roof:material)

`overture_roofShape_OSM_mapping.json` file contains mapping of OSM values to Overture `roofShape` property. OSM tags used for mapping in order:
- `roof:shape` (documentation: https://wiki.openstreetmap.org/wiki/Key:roof:shape)
- `building:roof:shape` (deprecated, documentation: https://wiki.openstreetmap.org/wiki/Key

File: task-force-docs/buildings/overture_roofMaterial_OSM_mapping.json [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    "plants",
+    "roof_greening"
+  ],
+  "gravel": [
+    "gravel"
+  ],
+  "metal": [
+    "metal",
+    "metal:sheet",
+    "zink",
```
[the remark this answers is not in the document]
Comment (LukaszMichalski-TomTom) [reply task-force-docs/buildings/overture_roofMaterial_OSM_mapping.json:32]: I will add `zinc` with 3 692 occurrences, but I will leave `zink` witch has 3 261.

[the remark this answers is not in the document]
Comment (ArthurTCrawford) [reply task-force-docs/buildings/overture_roofMaterial_OSM_mapping.json:33]: The German spelling is zink for zinc.

**Comment by vcschapp** [comment]
Reminder: @jwass took the action to consult with the PR authors and determine if this can be brought back to life.

**Review by jwass — DISMISSED** [review]

**Review by skmoore — APPROVED** [review]

**Review by vcschapp — APPROVED** [review]

**Review by TristanDiet-TomTom — DISMISSED** [review]

**Review by DavidKarlas — APPROVED** [review]

File: task-force-docs/buildings/overture_roofShape_OSM_mapping.json [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    "gabled_row",
+    "round_gabled",
+    "dutch_gabled",
+    "monopitch",
+    "2 faces (pitched)",
+    "gabeled",
+    "gabed",
+    "gambled",
+    "double_gabled",
+    "gabred",
```
[the remark this answers is not in the document]
Comment (vcschapp) [reply task-force-docs/buildings/overture_roofShape_OSM_mapping.json:28]: garbled?

File: task-force-docs/buildings/overture_facadeMaterial_OSM_mapping.json [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  "wood": [
+    "wood",
+    "reed",
+    "wattle_and_daub",
+    "timber_planks",
+    "wood/masonry",
+    "wood/bamboo",
+    "bamboo"
+  ]
+}
```
Comment (vcschapp) [inline task-force-docs/buildings/overture_facadeMaterial_OSM_mapping.json:86]: @LukaszMichalski-TomTom could you fix your tooling setup so it adds a newline at the end of the files?

File: task-force-docs/buildings/README.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# Description of the mappings file
+## facadeMaterial OSM mapping
+`overture_facadeMaterial_OSM_mapping.json` file contains mapping of OSM values to Overture `facedMaterial` property. OSM tags used for mapping in order:
```
Comment (vcschapp) [inline task-force-docs/buildings/README.md:3]: `facedMaterial` looks like a typo, should be `facadeMaterial` no?

**Review by vcschapp — CHANGES_REQUESTED** [review]
Could you please:

1. [x] Fix `facedMaterial` to `facadeMaterial`? (typo in the `README.md`)
2. [x] Ensure the files end in a newline (should not have the no newline at EOF icon at the bottom of every diff)

**Comment by jenningsanderson** [comment]
@LukaszMichalski-TomTom I updated these to be `snake_case` in-line with the new schema updates and added them all to one file.

These are out-of-sync with the current logic in the buildings airflow pipeline and schema, which is causing hundreds of thousands of errors.

Can you confirm that these are the correct mappings, and then I will update the logic here: https://github.com/OvertureMaps/tf-data-platform/blob/dev/datapipelines/airflow/dags/overture/query_fragments.py#L118 to reflect what's here.

**Comment by jenningsanderson** [comment]
Based on the current logic in the mapping JSON file (https://github.com/OvertureMaps/schema/pull/89/files#diff-5e18ae5feffd2d58657bf46be190bc986fb510fc7ee3b69f0e2d185828f4fad1), here are the counts for thee entire building theme: 

|roof_shape |_count|
|-------|---------|
| gabled | 4159376 |
| flat | 1370802 |
| hipped | 756957 |
| pyramidal | 249441 |
| skillion | 188343 |
| half_hipped | 100036 |
| saltbox | 55879 |
| round | 39348 |
| gambrel | 32996 |
| mansard | 23322 |
| dome | 16401 |
| onion | 5980 |
| sawtooth | 600 |
| spherical | 26 |


|roof_material | count|
|-------|---------|
| roof_tiles | 856423 |
| metal | 187222 |
| eternit | 129954 |
| concrete | 127903 |
| slate | 58635 |
| glass | 38825 |
| thatch | 21904 |
| grass | 12637 |
| plastic | 7316 |
| copper | 7299 |
| gravel | 7153 |
| wood | 3715 |
| solar_panels | 179 |


|facade_material | count|
|-------|---------|
| brick | 514852 |
| cement_block | 448061 |
| plaster | 398151 |
| wood | 210496 |
| concrete | 174449 |
| metal | 91201 |
| stone | 41789 |
| glass | 38035 |
| clay | 30661 |
| plastic | 16514 |
| timber_framing | 9975 |

**Comment by jenningsanderson** [comment]
I added `tar_paper` as an acceptable value because it's the 4th largest on OSM

**Review by vcschapp — APPROVED** [review]
```
    ) )        /\
   =====      /  \
  _|___|_____/ __ \____________
 |::::::::::/ |  | \:::::::::::|
 |:::::::::/  ====  \::::::::::|
 |::::::::/__________\:::::::::|
 |_________|  ____  |__________|
  | ______ | / || \ | _______ |
  ||  |   || ====== ||   |   ||
  ||--+---|| |    | ||---+---||
  ||__|___|| |   o| ||___|___||
  |========| |____| |=========|
 (^^-^^^^^-|________|-^^^--^^^)
 (,, , ,, ,/________\,,,, ,, ,)
','',,,,' /__________\,,,',',;;
```

**Review by DavidKarlas — APPROVED** [review]

**Pull request merged**